### PR TITLE
Toward dbe integration

### PR DESCRIPTION
**Pull request by anton-malakhov**

### CODE REVIEW

File: modin/experimental/engines/omnisci_on_ray/frame/partition_manager.py [hunk 1]
```diff
@@ -75,22 +77,24 @@ def run_exec_plan(cls, plan, index_cols, dtypes):
...
 
         calcite_plan = CalciteBuilder().build(plan)
         calcite_json = CalciteSerializer().serialize(calcite_plan)
 
-        sql = "execute relalg " + calcite_json
-        df = OmnisciServer()._worker._conn._execute(sql).to_df()
+        curs = omniSession.executeRA("execute relalg "+calcite_json) # TODO: remove prefix
+        rb = curs.getArrowRecordBatch()
+        assert rb
+        df = rb.to_pandas()  # TODO
```
Comment (Garra1980) [inline modin/experimental/engines/omnisci_on_ray/frame/partition_manager.py:88]: We need to get rid of this...

File: modin/experimental/engines/omnisci_on_ray/frame/partition.py [hunk 2]
```diff
@@ -44,8 +48,17 @@ def get(self):
...
             length=len(obj.index),
             width=len(obj.columns),
         )
-        
+
+    @classmethod
+    def put_arrow(cls, obj):
+        return OmnisciOnRayFramePartition(
+            object_id=ray.put(obj),
+            frame_id=OmnisciServer().put_arrow_to_omnisci(obj),  # TODO: make materialization later?
```
Comment (Garra1980) [inline modin/experimental/engines/omnisci_on_ray/frame/partition.py:60]: ...and this

Comment (anton-malakhov) [reply modin/experimental/engines/omnisci_on_ray/frame/partition.py:60]: sure thing

**Comment by ienkovich** [comment]
It's not just about intermediate step of file import. We are going to use Arrow format to transfer data between OmniSci and Modin in both directions. So we need to have Arrow import in a frame manager and Anton implemented it following from_pandas way. We don't have to add this interface into base classes and can leave it to our backend and engine only, but I think Arrow import is a nice thing to have.

File: modin/engines/base/frame/data.py [hunk 3]
```diff
@@ -1318,6 +1318,29 @@ def from_pandas(cls, df):
...
+
+        Args:
+            at: Arrow Table object.
+
+        Returns:
+            A new dataframe.
+        """
+        new_frame, new_lengths, new_widths = cls._frame_mgr_cls.from_arrow(at, True)
+        new_columns = at.column_names
+        new_index = None #new_columns[0].to_list() # TODO TOOOODOOO
```
Comment (anton-malakhov) [inline modin/engines/base/frame/data.py:1333]: @ienkovich

File: modin/experimental/engines/omnisci_on_ray/test/test_dataframe.py [hunk 4]
```diff
@@ -34,23 +34,24 @@ def run_and_compare(fn, data, data2=None, *args, **kwargs):
 
 
 class TestCSV:
-    root = os.path.abspath(__file__+"/.."*6)  # root of modin repo
+    root = os.path.abspath(__file__ + "/.." * 6)  # root of modin repo
     csv_file = os.path.join(root, "examples/data/boston_housing.csv")
     #  more csv files are here: modin/pandas/test/data
 
     def test_simple1(self):
```
Comment (Garra1980) [inline modin/experimental/engines/omnisci_on_ray/test/test_dataframe.py:41]: can we have some more meaningful name for test?

Comment (anton-malakhov) [reply modin/experimental/engines/omnisci_on_ray/test/test_dataframe.py:45]: TestCSVReader?

Comment (Garra1980) [reply modin/experimental/engines/omnisci_on_ray/test/test_dataframe.py:45]: test_read_csv, smth like that

Comment (anton-malakhov) [reply modin/experimental/engines/omnisci_on_ray/test/test_dataframe.py]: it needs PascalStyle if we want it to follow the suit

File: modin/experimental/engines/omnisci_on_ray/test/test_dataframe.py [hunk 5]
```diff
@@ -58,21 +58,18 @@ def test_housing_csv(self):
             ):
                 rp = pd.read_csv(csv_file, **kwargs)
                 rm = mpd.read_csv(csv_file, **kwargs)
-                print(rm)
-                # TODO: df_equals(rp, rm) does not support inexact comparison of numberic columns
+                df_equals(rp, rm, True)  # might need inexact comparison
```
Comment (Garra1980) [inline modin/experimental/engines/omnisci_on_ray/test/test_dataframe.py:61]: What exactly do we test here?

**Comment by anton-malakhov** [comment]
Replaced by https://github.com/intel-go/modin/pull/61

**Pull request closed without merging**